### ISSUE TITLE
The rule vue/no-unused-properties does not check refs in the template

### ISSUE THREAD

**Issue by danyadev** [vuejs/eslint-plugin-vue#1195]
**Tell us about your environment**	
* **ESLint version:** 7.2.0
* **eslint-plugin-vue version:** 7.0.0-alpha.5
* **Node version:**	v12.16.3

**Please show your full configuration:**

```js
module.exports = {
  root: true,

  parserOptions: {
    sourceType: 'module'
  },

  env: {
    browser: true,
    node: true,
    es2020: true
  },

  plugins: ['import'],

  settings: {
    'import/resolver': 'webpack',
    'import/ignore': []
  },

  extends: ['plugin:vue/vue3-recommended'],

  rules: {
    // *eslint rules*
    // *import rules*
    // *other vue rules*

    'vue/no-unused-properties': ['error', {
      groups: ['props', 'setup']
    }]
  }
};
```

**What did you do?**	

```vue
<template>
  <div ref="divRef">foo</div>
</template>

<script>
import { ref, onMounted } from 'vue';

export default {
  setup() {
    const divRef = ref(null);

    onMounted(() => {
      console.log(divRef);
    });

    return {
      divRef
    };
  }
};
</script>
```

**What did you expect to happen?**
No errors

**What actually happened?**
```
17:7  error  'divRef' of property returned from `setup()` found, but never used  vue/no-unused-properties
```

**Comment by ota-meshi** [comment]
Thank you for reporting these.

I will check and fix.
